Classes/common/Encryption/Keychain/CDTEncryptionKeychainManager.h

```diff
@@ -0,0 +1,76 @@
+//
+//  CDTEncryptionKeychainManager.h
+//  CloudantSync
+//
+//  Created by Enrique de la Torre Fernandez on 09/04/2015.
+//  Copyright (c) 2015 IBM Cloudant. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//  http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "CDTEncryptionKeychainStorage.h"
+
+/**
+ Use this class to generate a Data Protection Key (DPK), i.e. a strong password that can be used
+ later on for other purposes like ciphering a database.
+ 
+ The generated DPK is automatically encrypted and saved to the keychain, for this reason, it is
+ neccesary to provide a password to generate and retrieve the DPK.
+ */
+@interface CDTEncryptionKeychainManager : NSObject
+
+/**
+ Initialise a manager with a CDTEncryptionKeychainStorage instance.
+ 
+ A CDTEncryptionKeychainStorage gives access to the keychain throught an identifier, this means
+ that the DPK saved to the keychain is bound to this identifier. If a new manager is created with
+ a storage pointing to another identifier, the previous DPK will not be accesible. So you can
+ create as many DPKs as you want as long as you provide different identifiers.
```

Comment:
I think this is a bit confusing. Perhaps:

"""
A CDTEncryptionKeychainStorage binds an entry in the keychain to an identifier. The data protection key (DPK) saved to the keychain by this class will therefore be bound to the storage's identifier. To save different DPKs (say for different users of your application), create multiple managers using storages initialised with different identifiers.
"""

Is that correct? I think "the previous DPK will not be accessible" is a bit fear-inducing -- it's still accessible, you just need to use a storage with the right identifier.